### PR TITLE
Stub out initial user version detail page.

### LINKED ISSUE #4491: Stub user version detail page
1. Add a user version select to the item detail page.
  * The select only shows when the item has user versions and if the item only has a single user version that user version does not correspond with the current object version (i.e., don't provide a user version view of the current is that user version).
  * When the user selects a user version, the user version details page should show.
  * The user version select options should be formatted "Public version X" where X is the version number.
  * The placement of this select doesn't matter.
2. Stub the user version detail page.
  * The page should display the user version number.
  * The page should have a user version select.
  * The page should have a "Back to latest" button that returns to the item detail page.
  * The formatting of this page doesn't matter.

A feature flag isn't required. (There are no user versions in production.)

### PR DESCRIPTION
**Pull request by justinlittman**
closes #4491

# Why was this change made?
Support additional work on version detail page.

# How was this change tested?
Unit, QA

### CODE REVIEW

**Comment by justinlittman** [comment]
image

image

File: app/components/version_milestones_component.rb [hunk 1]
```diff
@@ -1,11 +1,28 @@
 # frozen_string_literal: true
 
 class VersionMilestonesComponent < ViewComponent::Base
-  def initialize(version:, title:, steps:)
+  def initialize(version:, milestones_presenter:)
     @version = version
```
Comment (mjgiarlo) [inline app/components/version_milestones_component.rb:5]: Is this instance variable no longer used? Seems like we use the version already set on the presenter currently, if I'm reading this correctly.

Comment (mjgiarlo) [reply app/components/version_milestones_component.rb:5]: Oh, nevermind. I see it's an attr_reader below!

File: app/controllers/catalog_controller.rb [hunk 2]
```diff
@@ -353,29 +353,36 @@ def lazy_wps_workflow_facet
...
+    @workflows = WorkflowService.workflows_for(druid: @druid)
+    milestones = MilestoneService.milestones_for(druid: @druid)
+    object_client = Dor::Services::Client.object(@druid)
     versions = object_client.version.inventory
+
     user_versions = object_client.user_version.inventory
-    @milestones_presenter = MilestonesPresenter.new(milestones:, versions:, user_versions:)
+    raise ActionController::RoutingError, 'Not Found' if @user_version && user_versions.none? { |version| version.userVersion.to_s == @user_version }
+
+    @milestones_presenter = MilestonesPresenter.new(milestones:, versions:, user_versions:, druid: @druid)
```
Comment (mjgiarlo) [inline app/controllers/catalog_controller.rb:380]: Take this suggestion or leave it: it looks like this method is now significantly more complex, and the milestones presenter is still on the slim side. If you were interested in re-balancing the complexity, I think you could shovel much of lines 372-377 into the presenter itself, who can take more responsibility for its own setup, given `@druid` and `@user_version`?

File: app/components/version_milestones_component.rb (hunk 1 above)
Comment (justinlittman) [reply app/components/version_milestones_component.rb:5]: Version is the system version. User version is from the presenter. Sigh.

File: app/controllers/catalog_controller.rb (hunk 2 above)
Comment (justinlittman) [reply app/controllers/catalog_controller.rb:380]: That makes sense. Would you be OK if I did this in a follow-up PR? I want to unblock some of the additional work.

Comment (mjgiarlo) [reply app/controllers/catalog_controller.rb:380]: @justinlittman 💯

**Review by mjgiarlo — APPROVED** [review]

**Pull request merged**